siliconcompiler/report/Home.py

```diff
@@ -0,0 +1,601 @@
+            node['label'] = folder
+            node['value'] = path_name
+            node['children'] = children
+            subsect_logs_and_reports[folder] = node
+
+
+def get_nodes_and_edges(chip, node_dependencies, successful_path):
+    """
+    Returns the nodes and edges required to make a streamlit_agraph.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        node_dependencies (dict) : dictionary where the values of the keys are
+                                   the edges
+        successful_path (set) : contains all the nodes that are part of the
+                                'winning' path
+    """
+    nodes = []
+    edges = []
+    opacity = 0.2
+    width = 1
+    for step, index in node_dependencies:
+        if (step, index) in successful_path:
+            opacity = 1
+            if (step, index) in chip._get_flowgraph_exit_nodes() or \
+               (step, index) in chip._get_flowgraph_entry_nodes():
+                width = 3
+
+        flow = chip.get("option", "flow")
+        task_status = chip.get('flowgraph', flow, step, index, 'status')
+        if task_status == TaskStatus.SUCCESS:
+            node_color = success_color
+        elif task_status == TaskStatus.ERROR:
+            node_color = failure_color
+        else:
+            node_color = pending_color
+
+        tool, task = chip._get_tool_task(step, index)
+        label = step + index + "\n" + tool + "/" + task
+        if chip._is_builtin(tool, task):
+            label = step + index + "\n" + tool
+
+        nodes.append(Node(id=step + index,
+                          label=label,
+                          color=node_color,
+                          opacity=opacity,
+                          borderWidth=width,
+                          shape='oval'))
+        width = 3
+        for source_step, source_index in node_dependencies[step, index]:
+            if (source_step, source_index) in successful_path and \
+               (source_step, source_index) in successful_path:
+                width = 5
+            edges.append(Edge(source=source_step + source_index,
+                              dir='up',
+                              target=step + index,
+                              width=width,
+                              color='black',
+                              curve=True))
+
+    return nodes, edges
+
+
+def show_file_preview(display_file_content):
+    """
+    Displays the file_preview if present. If not, displays an error message.
+
+    Args:
+        display_file_content (bool) : boolean representing if we should (true)
+                                      or should not (false) display content
+    """
+    if display_file_content:
+        path = streamlit.session_state['selected'][0]
+        file_name, file_extension = os.path.splitext(path)
+        header_col, download_col = streamlit.columns([0.89, 0.11], gap='small')
```

Comment:
magic numbers, how did they come about and should we make those inputs?

Reply:
They probably should not be inputs. All widgets are left-aligned and I found that when I made download_col the right 11% of the screen, the download button was the farthest right it could go

Reply:
how does this scale with screensize then?